ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward.hpp

```diff
@@ -0,0 +1,82 @@
+
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "device/binary_backward_op.cpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/data_movement.hpp"
+#include "tt_eager/tt_dnn/op_library/run_operation.hpp"
+
+namespace ttnn {
+
+namespace operations::binary_backward {
+
+template <BinaryBackwardOpType binary_backward_op_type>
+struct ExecuteBinaryBackward {
+
+    static inline const std::array<TensorSchema, 3> input_tensor_schemas() {
+        return {
+            ttnn::TensorSchema{
+                2,
+                4,
+                {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b, ttnn::uint16},
+                {ttnn::TILE_LAYOUT},
+                true,
+                false,
+                false,
+                false},
+            ttnn::TensorSchema{
+                2,
+                4,
+                {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b, ttnn::uint16},
+                {ttnn::TILE_LAYOUT},
+                true,
+                false,
+                false,
+                false},
+            ttnn::TensorSchema{
+                2,
+                4,
+                {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b, ttnn::uint16},
+                {ttnn::TILE_LAYOUT},
+                true,
+                false,
+                false,
+                false}};
+    }
+
+    template <typename... Args>
+    static auto input_tensors_to_validate(const Tensor &grad_tensor, const Tensor &input_tensor_a, const Tensor &input_tensor_b, Args &&...args) {
+        return std::forward_as_tuple(grad_tensor, input_tensor_a, input_tensor_b);
+    }
+
+    static std::vector<Tensor> execute_on_worker_thread(
+        const Tensor &grad_tensor_arg,
+        const Tensor &input_tensor_a_arg,
+        const Tensor &input_tensor_b_arg,
+        const MemoryConfig &memory_config) {
+
+        auto op_type = utils::get_function(binary_backward_op_type);
+
+        return tt::tt_metal::operation::decorate_as_composite(__func__, op_type)(grad_tensor_arg, input_tensor_a_arg, input_tensor_b_arg, memory_config);
+        }
+
+    static inline std::vector<Tensor> create_async_output_tensors(
```

Comment:
@arakhmati is this the legit part of ttnn infra?

Reply:
@ayerofieiev-tt , we found a way to do this without `decorate_as_composite` function, will push it in further commits

Reply:
Work in progress on branch [Aswinmcw/binary_backward_optional](https://github.com/tenstorrent/tt-metal/compare/virdhatchani/backward_binary_merge...Aswinmcw/binary_backward_optional?expand=1)